src/java/io/bazel/rulesscala/worker/GenericWorker.java

```diff
@@ -1,117 +0,0 @@
-package io.bazel.rulesscala.worker;
-
-import static java.nio.charset.StandardCharsets.UTF_8;
-
-import com.google.devtools.build.lib.worker.WorkerProtocol.WorkRequest;
-import com.google.devtools.build.lib.worker.WorkerProtocol.WorkResponse;
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.io.PrintStream;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.util.Arrays;
-import java.util.List;
-
-public class GenericWorker {
-  protected final Processor processor;
-
-  public GenericWorker(Processor p) {
-    processor = p;
-  }
-
-  protected void setupOutput(PrintStream ps) {
-    System.setOut(ps);
-    System.setErr(ps);
-  }
-
-  // Mostly lifted from bazel
-  private void runPersistentWorker() throws IOException {
-    PrintStream originalStdOut = System.out;
-    PrintStream originalStdErr = System.err;
-
-    while (true) {
-      try {
-        WorkRequest request = WorkRequest.parseDelimitedFrom(System.in);
-        if (request == null) {
```

Comment:
Why drop this?

Reply:
I don't think this ever happens. Bazel has workers that skip this check: https://github.com/bazelbuild/bazel/blob/87f19705026e90e774ce6773299429b8267ecfe8/src/java_tools/singlejar/java/com/google/devtools/build/singlejar/SingleJarWorker.java#L56-L66

Happy to add it back, though.